exonum-java-binding/time-oracle/src/main/java/com/exonum/binding/time/TimeSchemaProxy.java

```diff
@@ -71,18 +72,18 @@ private static boolean isTimeOracleInstance(ArtifactId artifactId) {
 
   @Override
   public ProofEntryIndexProxy<ZonedDateTime> getTime() {
-    return ProofEntryIndexProxy.newInstance(
-        indexName(TimeIndex.TIME), view, ZONED_DATE_TIME_SERIALIZER);
+    IndexAddress address = indexAddress(TimeIndex.TIME);
+    return access.getProofEntry(address, ZONED_DATE_TIME_SERIALIZER);
   }
 
   @Override
   public ProofMapIndexProxy<PublicKey, ZonedDateTime> getValidatorsTimes() {
-    return ProofMapIndexProxy.newInstanceNoKeyHashing(indexName(TimeIndex.VALIDATORS_TIMES), view,
-        PUBLIC_KEY_SERIALIZER, ZONED_DATE_TIME_SERIALIZER);
+    IndexAddress address = indexAddress(TimeIndex.VALIDATORS_TIMES);
+    return access.getRawProofMap(address, PUBLIC_KEY_SERIALIZER, ZONED_DATE_TIME_SERIALIZER);
   }
 
-  private String indexName(String simpleName) {
-    return name + "." + simpleName;
+  private IndexAddress indexAddress(String simpleName) {
+    return IndexAddress.valueOf(name + "." + simpleName);
```

Comment:
nit: It is actually against IndexAddress documentation, as `.` is not allowed 🤔

Reply:
Thanks, fixed: 3a1f2ea ('.' — allowed, e.g., in non-prefixed views).